syft/formats/common/cyclonedxhelpers/decoder.go

```diff
@@ -23,8 +26,14 @@ func GetValidator(format cyclonedx.BOMFileFormat) sbom.Validator {
 			return err
 		}
 		// random JSON does not necessarily cause an error (e.g. SPDX)
-		if (cyclonedx.BOM{} == *bom || ((format == cyclonedx.BOMFileFormatXML && bom.XMLNS == "") || bom.Components == nil)) {
-			return fmt.Errorf("not a valid CycloneDX document")
+		if format == cyclonedx.BOMFileFormatXML {
+			if (!strings.Contains(bom.XMLNS, cycloneDXXmlSchema) || cyclonedx.BOM{} == *bom) {
+				return fmt.Errorf("not a valid CycloneDX document")
+			}
+		} else {
+			if (bom.Components == nil || cyclonedx.BOM{} == *bom) {
+				return fmt.Errorf("not a valid CycloneDX document")
+			}
```

Comment:
Having nested `if` statements like this duplicates some of the logic and makes things a little harder to reason about. What about a change like this?
```suggestion
		xmlWithoutNS := format == cyclonedx.BOMFileFormatXML && !strings.Contains(bom.XMLNS, cycloneDXXmlSchema)
		if cyclonedx.BOM{} == *bom || bom.Components == nil || xmlWithoutNS {
			return fmt.Errorf("not a valid CycloneDX document")
		}
```